sphinx_rtd_theme/versions.html

```diff
@@ -1,34 +1,47 @@
-{% if READTHEDOCS %}
-{# Add rst-badge after rst-versions for small badge style. #}
-  <div class="rst-versions" data-toggle="rst-versions" role="note" aria-label="{{ _('Versions') }}">
+<script async type="text/javascript" src="http://localhost:8000/readthedocs-addons.js"></script>
+<script type="text/javascript">
+ document.addEventListener("readthedocs-addons-data-ready", function(event) {
+     const config = event.detail;
+
+     const flyout = `
```

Comment:
Injecting HTML as a string template does work, however eventually I'd like to give theme authors a stronger example to work from. This does remove translated strings by pushing them into JS (it is possible to translate the strings in Python via template strings embedded in JS, but this will look especially gnarly).

That's not to say that I think this needs to change, but when we go to point users towards an example they should have something more standard to work from, like an example using `document.createElement` to populate a dropdown perhaps.

Reply:
> What do you mean with a "stronger example to work from"?

Using string interpolation to inject a blob of HTML like this is certainly the *fastest* solution. But because of that might this pattern not be super usable or appealing for serious applications. This isn't a commonly used pattern as it's easier to author HTML outside JS entirely, similar to how I'd want to use Jinja instead of Python string interpolation. Lit is special, as the entire implementation is in JS already.

A stronger example could be using JS to provide data to existing HTML structures defined in templates. For example, populating a version selector dropdown. Personally, if I wanted a custom flyout in a theme, I would author the HTML structure in my templates and hide the custom flyout until the data event fired -- instead of creating the whole structure in JS strings.

Again though, I don't think we need to reshape things here for our testing. But I also probably would want to create a more real scenario for our example rather than build on top of this pattern going forward.

> We do not have too many strings to translate.

I wasn't necessarily looking for a solution for our code, my point was more that this is a pattern that will complicate development for theme maintainers. Theme maintainers are already doing translations in their template and we should meet them where they are.

> What are the benefits of this over a string template?

The biggest benefit is separation between layers. HTML authoring is in a single place, not split between JS and HTML template code. JS is treated like more of a data layer in this case. This is the main draw of data binding libraries like Vue, Knockout, and friends.

I don't think we need/want to show an example using data binding, so we will need to add some elements with JS. This is fine, it's just creating a large nested blob of HTML in JS that I'd avoid.

It also doesn't require theme maintainers have new patterns for localization, etc.

Reply:
I followed the pattern we already have here to manage translations `{{ _("String to translate") }}` in 04d79091091e0a9289ca274bb568dadda8164f68. It works fine:

![Screenshot_2024-04-15_15-55-54](https://github.com/readthedocs/sphinx_rtd_theme/assets/244656/844f9c4b-8ecd-4502-a07e-aa7e17393939)